secret/secrets_manager_vault_test.go

```diff
@@ -0,0 +1,133 @@
+package secret
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/evergreen-ci/cocoa/awsutil"
+	"github.com/evergreen-ci/utility"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSecretsManager(t *testing.T) {
+	assert.Implements(t, (*Vault)(nil), &BasicSecretsManager{})
+
+}
+
+func TestVaultCreateAndDeleteSecret(t *testing.T) {
+	checkAWSEnvVars(t)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
```

Comment:
Same thing as above about `context.WithCancel` instead of `context.WithTimeout`.